nucliadb/nucliadb/search/api/v1/resource/ask.py

```diff
@@ -0,0 +1,139 @@
+from nucliadb.models.responses import HTTPClientError
+from nucliadb.search import SERVICE_NAME, logger
+from nucliadb.search.api.v1.router import KB_PREFIX, api
+from nucliadb.search.predict import SendToPredictError
+from nucliadb.search.search.exceptions import ResourceNotFoundError
+from nucliadb.search.utilities import get_predict
+from nucliadb_models.resource import NucliaDBRoles
+from nucliadb_models.search import AskRequest, AskResponse, TextBlocks
+from nucliadb_utils.authentication import requires
+from nucliadb_utils.exceptions import LimitsExceededError
+from nucliadb_utils.utilities import get_storage
+
+
+@api.post(
+    f"/{KB_PREFIX}/{{kbid}}/resource/{{rid}}/ask",
+    status_code=200,
+    name="Ask a Resource",
+    summary="Ask a question to a resource",
+    description="Ask to the complete content of the resource",
+    tags=["Search"],
+)
+@requires(NucliaDBRoles.READER)
+@version(1)
+async def resource_ask_endpoint(
+    request: Request,
+    response: Response,
+    kbid: str,
+    rid: str,
+    item: AskRequest,
+) -> Union[AskResponse, HTTPClientError]:
+    try:
+        return await resource_ask(kbid, rid, item)
+    except ResourceNotFoundError:
+        return HTTPClientError(status_code=404, detail="Resource not found")
+    except LimitsExceededError as exc:
+        return HTTPClientError(status_code=exc.status_code, detail=exc.detail)
+    except SendToPredictError:
+        return HTTPClientError(status_code=503, detail="Ask service not available")
+
+
+async def resource_ask(
+    kbid: str,
+    rid: str,
+    item: AskRequest,
+) -> AskResponse:
+    blocks = await get_resource_text_blocks(kbid, rid)
+
+    predict = get_predict()
+    answer = await predict.ask_document(kbid, item.question, blocks)
+
+    return AskResponse(answer=answer)
+
+
+async def get_resource_text_blocks(kbid: str, rid: str) -> TextBlocks:
+    """
+    Iterate over all fields of the resource and get its extracted text.
+    Slice file extracted texts by paragraphs.
+    """
+    blocks = []
+    driver = get_driver()
+    storage = await get_storage(service_name=SERVICE_NAME)
+    async with driver.transaction() as txn:
+        kb = KnowledgeBox(txn, storage, kbid)
+        orm_resource = await kb.get(rid)
+        if orm_resource is None:
+            raise ResourceNotFoundError()
+
+        for field_type, field_id in await orm_resource.get_fields_ids():
+            field_obj = await orm_resource.get_field(field_id, field_type, load=False)
+            etxt = await field_obj.get_extracted_text()
+            if etxt is None:
+                logger.warning(
+                    f"Skipping field {field_id}, as it does not have extracted text yet!"
+                )
+                continue
+
+            fcm = await field_obj.get_field_metadata()
+            if fcm is None:
+                logger.warning(f"Field metadata not found for {field_id}")
+                blocks.append(split_by_newlines(etxt))
```

Comment:
At the very least, maybe we don't do this manual splitting here and just send add the block without newline split?